link/uprobe_test.go

```diff
@@ -293,3 +325,61 @@ func TestUprobeProgramCall(t *testing.T) {
 	// Assert that this time the value has not been updated.
 	assertMapValue(t, m, 0, 0)
 }
+
+func TestUprobeProgramCallSharedlib(t *testing.T) {
+	m, p := newUpdaterMapProg(t, ebpf.Kprobe)
+
+	// External symbol.
+	sym := "open"
+
+	// Load the '/bin/bash' executable.
+	ex, err := OpenExecutable("/bin/bash")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Obtain external symbol address.
+	out, err := exec.Command("/usr/bin/gdb", "/bin/bash", "--batch", "-ex", `printf "%p", `+sym).Output()
```

Comment:
out of curiosity, could you try running `objdump -d /bin/bash |rg 'open@plt'` on your arch machine?

Reply:
```
λ  ~  objdump -d /bin/bash | grep '<open@'
   211b1:	ff 15 19 4c 0c 00    	call   *0xc4c19(%rip)        # e5dd0 <open@GLIBC_2.2.5>
   2170d:	ff 15 bd 46 0c 00    	call   *0xc46bd(%rip)        # e5dd0 <open@GLIBC_2.2.5>
   30832:	ff 15 98 55 0b 00    	call   *0xb5598(%rip)        # e5dd0 <open@GLIBC_2.2.5>
   30874:	ff 15 56 55 0b 00    	call   *0xb5556(%rip)        # e5dd0 <open@GLIBC_2.2.5>
<snip
```
Only references to `open@GLIBC_2.2.5`, no `open@plt` in the output.

Reply:
at this point, regarding the CI integration test, I think we can only choose between these 3 options:

1) depend on `.plt` stub/trampoline being present on the CI runners' ELF and gdb being available
2) attach the uprobe to the shared lib directly (https://github.com/iovisor/bcc/blob/master/libbpf-tools/uprobe_helpers.c#L51). Additional questions: parse lib version from elf? hardcode the lib path? 
3) skip the integration test completely as only the offset changes

m2c: I think both `1.` and `3.` are ok while `2.` may be a little overkill for the scope of the test

thoughts?

Reply:
Both 1. and 3. are OK for me, up to Timo. Regarding option two: is that the way forward for attaching uprobe to shared libs?

Reply:
I had in mind to use the `.plt` section but apparently it is not always present. `2.` should be another way to do that automatically, **but** a pid must be provided, else every other process using the shared lib will trigger program calls

Reply:
/ping @ti-mo - any thoughts/suggestions on this?

Reply:
@lmb I've discussed with @ti-mo on Slack and, if I understood correctly, we agreed that testing this without using something like gdb would be overkill. I've removed the integration test for now, I think it will be easier to implement once there will be something in place for taking the offset automatically

Reply:
Sounds good :) Thank you for sticking with this for so long!